packages/devtools_app/lib/src/screens/deep_link_validation/deep_links_screen.dart

```diff
@@ -24,8 +28,144 @@ class DeepLinksScreen extends Screen {
 
   @override
   Widget build(BuildContext context) {
-    return const Center(
-      child: Text('TODO: build deep link validation tool'),
+    return const DeepLinkPage();
+  }
+}
+
+class DeepLinkPage extends StatefulWidget {
+  const DeepLinkPage({super.key});
+
+  @override
+  State<DeepLinkPage> createState() => _DeepLinkPageState();
+}
+
+class _DeepLinkPageState extends State<DeepLinkPage>
+    with
+        AutoDisposeMixin,
+        SingleTickerProviderStateMixin,
+        ProvidedControllerMixin<DeepLinksController, DeepLinkPage> {
+  @override
+  void didChangeDependencies() {
+    super.didChangeDependencies();
+    if (!initController()) return;
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return ListView(
+      children: [
+        Row(
+          children: [
+            Expanded(
+              child: Text(
+                'All deep links',
+                style: Theme.of(context).textTheme.bodyLarge,
+              ),
+            ),
+            InkWell(
+              onTap: () {
+                controller.bundleByDomain = !controller.bundleByDomain;
+              },
+              child: const Text('Bundle by domain'),
+            ),
+            const SizedBox(width: denseSpacing),
+            SearchBar(
+              leading: const Icon(Icons.search),
+              onChanged: (value) {
+                controller.searchContent = value;
+              },
+              constraints: BoxConstraints.tight(
+                Size(defaultSearchFieldWidth, defaultTextFieldHeight),
+              ),
+            ),
+          ],
+        ),
+        const SizedBox(height: denseSpacing),
+        ValueListenableBuilder<List<LinkData>>(
+          valueListenable: controller.linkDatasNotifier,
+          builder: (context, linkDatas, _) => _DataTable(
+            linkDatas: linkDatas,
+            bundleByDomain: controller.bundleByDomain,
+          ),
+        ),
+      ],
+    );
+  }
+}
+
+class _DataTable extends StatelessWidget {
+  const _DataTable({required this.bundleByDomain, required this.linkDatas});
+  final bool bundleByDomain;
+  final List<LinkData> linkDatas;
+
+  @override
+  Widget build(BuildContext context) {
+    final List<DataRow> rows = [
+      for (var i = 0; i < linkDatas.length; i++)
```

Comment:
will `linkDatas` be large? We should be careful about building the entire table if there will be a lot of data here, as this could cause performance issues. If the data set is large, we should build this table in a lazy way.

Reply:
Thank you! I've updated it to use the FlatTable

Reply:
some large company may have ~1000 deeplink.